src/stan_math_backend/Stan_math_code_gen.ml

```diff
@@ -657,12 +680,53 @@ let pp_register_map_rect_functors ppf p =
     (list ~sep:cut pp_register_functor)
     (List.mapi ~f:(fun i f -> (i + 1, f)) functors)
 
+let fun_used_in_reduce_sum p =
+  let rec find_functors_expr accum Expr.Fixed.({pattern; _}) =
+    match pattern with
+    | FunApp (StanLib, "reduce_sum", {pattern= Var f; _} :: _) -> f :: accum
+    | FunApp (StanLib, _, fk) -> 
+    List.concat [List.concat (List.map ~f:(fun i -> (find_functors_expr String.Set.empty i)) fk);accum]
+    | _ -> accum
```

Comment:
Doesn't look like you "bailed on the sets thing", there's still a String.Set.empty here. More importantly, I don't think you don't need this FunApp branch. The last wildcard case should be
```ocaml
    | x -> Expr.Fixed.Pattern.fold find_functors_expr accum x
```
and covers everything.

Reply:
Fantastic. Thanks for all the tips! I redid this to sets now.